src/spatialdata/models/models.py

```diff
@@ -792,3 +869,59 @@ def _validate_and_return(
     if isinstance(e, AnnData):
         return _validate_and_return(TableModel, e)
     raise TypeError(f"Unsupported type {type(e)}")
+
+
+def get_table_keys(table: AnnData) -> tuple[str | list[str], str, str]:
+    """
+    Get the table keys giving information about what spatial element is annotated.
+
+    The first element returned gives information regarding which spatial elements are annotated by the table, the second
+    element gives information which column in table.obs contains the information which spatial element is annotated
+    by each row in the table and the instance key indicates the column in obs giving information of the id of each row.
+
+    Parameters
+    ----------
+    table:
+        AnnData table for which to retrieve the spatialdata_attrs keys.
+
+    Returns
+    -------
+    None | tuple[str | list[str], str, str]
+        The keys in table.uns['spatialdata_attrs']
+    """
+    if table.uns.get(TableModel.ATTRS_KEY):
+        attrs = table.uns[TableModel.ATTRS_KEY]
+        return attrs[TableModel.REGION_KEY], attrs[TableModel.REGION_KEY_KEY], attrs[TableModel.INSTANCE_KEY]
+
+    raise ValueError("No spatialdata_attrs key found in table.uns. Therefore, no table keys found.")
+
+
+def check_target_region_column_symmetry(table: AnnData, region_key: str, target: str | pd.Series) -> None:
+    """
+    Check region and region_key column symmetry.
+
+    This checks whether the specified targets are also present in the region key column in obs and raises an error
+    if this is not the case.
+
+    Parameters
+    ----------
+    table
+        Table annotating specific SpatialElements
+    region_key
+        The column in obs containing for each row which SpatialElement is annotated by that row.
+    target
+         Name of target(s) SpatialElement(s)
+
+    Raises
+    ------
+    ValueError
+        If the regions specified in table.uns["spatialdata_attrs"] are not present in the region column of table.obs.
+    """
+    found_regions = set(table.obs[region_key].unique().tolist())
+    target_element_set = [target] if isinstance(target, str) else target
+    symmetric_difference = found_regions.symmetric_difference(target_element_set)
+    if symmetric_difference:
+        raise ValueError(
```

Comment:
Also here I think a warning instead of an error would be ok, but I would like to experiment a bit more with the PR before changing this one. So I'd merge it with this like this, and eventually relax.